src/main/java/org/jboss/forge/arquillian/command/cube/CubeSetupCommand.java

```diff
@@ -0,0 +1,336 @@
+import org.jboss.forge.addon.ui.result.Result;
+import org.jboss.forge.addon.ui.result.Results;
+import org.jboss.forge.addon.ui.util.Categories;
+import org.jboss.forge.addon.ui.util.Metadata;
+import org.jboss.forge.arquillian.api.ArquillianFacet;
+import org.jboss.forge.arquillian.api.TestFrameworkFacet;
+import org.jboss.forge.arquillian.api.YamlGenerator;
+import org.jboss.forge.arquillian.api.cube.CubeSetupFacet;
+import org.jboss.forge.arquillian.container.model.CubeConfiguration;
+import org.jboss.forge.furnace.util.Strings;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.jboss.forge.arquillian.util.StringUtil.getStringForCLIDisplay;
+
+public class CubeSetupCommand extends AbstractProjectCommand implements UICommand {
+
+    private CubeSetupFacet cubeSetupFacet;
+
+    @Inject
+    private ProjectFactory projectFactory;
+
+    @Inject
+    private FacetFactory facetFactory;
+
+    @Inject
+    private ResourceFactory resourceFactory;
+
+    @Inject
+    @WithAttributes(shortName = 't', label = "Type", type = InputType.DROPDOWN)
+    private UISelectOne<String> type;
+
+    @Inject
+    @WithAttributes(shortName = 'f', label = "File", required = true)
+    private UIInput<String> filePath;
+
+    @Inject
+    @WithAttributes(shortName = 'm', label = "Docker Machine", type = InputType.CHECKBOX)
+    private UIInput<Boolean> dockerMachine;
+
+    @Inject
+    @WithAttributes(shortName = 'n', label = "Docker Machine Name")
+    private UIInput<String> machineName;
+
+    @Inject
+    @WithAttributes(shortName = 'd', label = "Docker File Name")
+    private UIInput<String> dockerFileName;
+
+
+    @Override
+    public void initializeUI(UIBuilder builder) throws Exception {
+        builder
+            .add(type)
+            .add(filePath)
+            .add(dockerMachine)
+            .add(machineName)
+            .add(dockerFileName);
+
+        final List<String> types = Arrays.stream(CubeConfiguration.values())
+            .map(CubeConfiguration::getType)
+            .collect(Collectors.toList());
+
+        type.setValueChoices(types);
+        type.setEnabled(true);
+        type.setRequired(() -> true);
+        type.setItemLabelConverter(source -> {
+            if (source == null) {
+                return null;
+            }
+            if (builder.getUIContext().getProvider().isGUI()) {
+                return source;
+            }
+            return getStringForCLIDisplay(source);
+        });
+
+        filePath.setEnabled(true);
+
+        dockerMachine.setEnabled(this::isDockerOrDockerCompose);
+        dockerMachine.setRequired(this::isDockerOrDockerCompose);
+
+        machineName.setEnabled(this::isDockerMachine);
+        machineName.setRequired(this::isDockerMachine);
```

Comment:
I will not add this flag as required since Linux developers should be set every to none.